Web Inspector: Make frame targets compatible with WebKitLegacy

the-chenergy opened the pull request:
#### 5f614bee7105d19f74d3322b9d15ff62e218a442
<pre>
Web Inspector: Make frame targets compatible with WebKitLegacy
<a href="https://bugs.webkit.org/show_bug.cgi?id=302125">https://bugs.webkit.org/show_bug.cgi?id=302125</a>
<a href="https://rdar.apple.com/problem/164203459">rdar://problem/164203459</a>

Reviewed by BJ Burg.

With our plan to slowly move inspector agent implementations from the
page target + InspectorController to the frame target + FrameInspectorController
for site isolation, we run into the issue that inspecting WK1 uses a WI.DirectBackendTarget
to directly contact the InspectorController, so any domains moving out
of the page will become opaque to that DirectBackendTarget.

To solve this compatibility issue and hopefully prevent having to leave
copies of any old agent code in the page that we must still maintain,
introduce a multi-target support to WK1 so that inspector frontend can
use WI.MultiplexingBackendTarget to inspect either WK1 or WK2. This is
done by adding LegacyWebPage{Debuggable,InspectorController} acting as
destinations of &quot;web-page&quot;-typed debuggable and target in WK1, which
will route commands and events to the appropriate targets just like in
WK2, except without using IPC.

Test: re-enabled target-manager.html for WK1. No new tests otherwise;
inspection of WebKitLegacy should have no observable difference.

* LayoutTests/platform/mac-wk1/TestExpectations:
* Source/JavaScriptCore/inspector/remote/RemoteControllableTarget.h:
* Source/JavaScriptCore/inspector/remote/cocoa/RemoteInspectorCocoa.mm:
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorSocket.cpp:
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/WebCore/inspector/InspectorBackendClient.h:
(WebCore::InspectorBackendClient::allowRemoteInspectionToPageDirectly const): Deleted.
* Source/WebCore/inspector/InspectorFrontendClientLocal.h:
* Source/WebCore/page/Page.cpp:
(WebCore::m_mediaSessionManagerFactory):
* Source/WebKitLegacy/WebCoreSupport/LegacyWebPageDebuggable.cpp: Added.
(LegacyWebPageDebuggable::LegacyWebPageDebuggable):
(LegacyWebPageDebuggable::create):
(LegacyWebPageDebuggable::name const):
(LegacyWebPageDebuggable::url const):
(LegacyWebPageDebuggable::hasLocalDebugger const):
(LegacyWebPageDebuggable::connect):
(LegacyWebPageDebuggable::disconnect):
(LegacyWebPageDebuggable::dispatchMessageFromRemote):
(LegacyWebPageDebuggable::setIndicating):
(LegacyWebPageDebuggable::setNameOverride):
(LegacyWebPageDebuggable::detachFromPage):
* Source/WebKitLegacy/WebCoreSupport/LegacyWebPageDebuggable.h: Added.
* Source/WebKitLegacy/WebCoreSupport/LegacyWebPageInspectorController.cpp: Added.
(PageTarget::identifier):
(PageTarget::PageTarget):
(FrameTarget::identifier):
(FrameTarget::FrameTarget):
(LegacyWebPageInspectorController::create):
(LegacyWebPageInspectorController::LegacyWebPageInspectorController):
(LegacyWebPageInspectorController::frameCreated):
(LegacyWebPageInspectorController::frameDestroying):
(LegacyWebPageInspectorController::pageDestroying):
(LegacyWebPageInspectorController::addTarget):
(LegacyWebPageInspectorController::removeTarget):
(LegacyWebPageInspectorController::connectFrontend):
(LegacyWebPageInspectorController::disconnectFrontend):
(LegacyWebPageInspectorController::hasLocalFrontend const):
(LegacyWebPageInspectorController::disconnectAllFrontends):
(LegacyWebPageInspectorController::dispatchMessageFromFrontend):
(LegacyWebPageInspectorController::sendMessageToInspectorFrontend):
* Source/WebKitLegacy/WebCoreSupport/LegacyWebPageInspectorController.h: Added.
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/cf/WebCoreSupport/WebInspectorClientCF.cpp:
(WebInspectorClient::sendMessageToFrontend):
* Source/WebKitLegacy/ios/WebCoreSupport/WebInspectorClientIOS.mm:
(WebInspectorFrontendClient::WebInspectorFrontendClient):
* Source/WebKitLegacy/mac/WebCoreSupport/WebInspectorClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebInspectorClient.mm:
(WebInspectorClient::openLocalFrontend):
(WebInspectorFrontendClient::WebInspectorFrontendClient):
(WebInspectorFrontendClient::sendMessageToBackend):
(-[WebInspectorWindowController destroyInspectorView]):
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(+[WebFrame _createFrameWithPage:frameName:frameView:ownerElement:]):
(+[WebFrame _createMainFrameWithPage:frameName:frameView:]):
(-[WebFrame _clearCoreFrame]):
* Source/WebKitLegacy/mac/WebView/WebFrameInternal.h:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _commonInitializationWithFrameName:groupName:]):
(-[WebView initSimpleHTMLDocumentWithStyle:frame:preferences:groupName:]):
(-[WebView _close]):
(-[WebView allowsRemoteInspection]):
(-[WebView setAllowsRemoteInspection:]):
(-[WebView inspectorController]):
* Source/WebKitLegacy/mac/WebView/WebViewData.h:
* Source/WebKitLegacy/mac/WebView/WebViewInternal.h:

Canonical link: <a href="https://commits.webkit.org/304467@main">https://commits.webkit.org/304467@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/536534e136eacf41853ac4267692b6d0986fed5d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134836 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7272 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/46129 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/142352 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/86728 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/f2ddc900-26d4-4022-b4f2-ee6ea3c0461d) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/136706 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/7884 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/7119 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103033 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/70326 "Passed tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137783 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5568 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/120896 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83868 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/370724e2-6524-4720-b48f-3790a493ce4f) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5399 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3014 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/2943 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/126872 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/114614 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/39050 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/145048 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/133343 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/6947 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/39625 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/111415 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7007 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/5945 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111752 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5238 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117177 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/60878 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20909 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/6993 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35351 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/166243 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/6779 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/70573 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/43449 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/7015 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/6889 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

dcrousso reviewed (changes requested):
please see <https://github.com/WebKit/WebKit/pull/53561#pullrequestreview-3431082999>

the-chenergy committed:
Web Inspector: Make frame targets compatible with WebKitLegacy
https://bugs.webkit.org/show_bug.cgi?id=302125
rdar://problem/164203459

Reviewed by BJ Burg.

With our plan to slowly move inspector agent implementations from the
page target + InspectorController to the frame target + FrameInspectorController
for site isolation, we run into the issue that inspecting WK1 uses a WI.DirectBackendTarget
to directly contact the InspectorController, so any domains moving out
of the page will become opaque to that DirectBackendTarget.

To solve this compatibility issue and hopefully prevent having to leave
copies of any old agent code in the page that we must still maintain,
introduce a multi-target support to WK1 so that inspector frontend can
use WI.MultiplexingBackendTarget to inspect either WK1 or WK2. This is
done by adding LegacyWebPage{Debuggable,InspectorController} acting as
destinations of "web-page"-typed debuggable and target in WK1, which
will route commands and events to the appropriate targets just like in
WK2, except without using IPC.

Test: re-enabled target-manager.html for WK1. No new tests otherwise;
inspection of WebKitLegacy should have no observable difference.

* LayoutTests/platform/mac-wk1/TestExpectations:
* Source/JavaScriptCore/inspector/remote/RemoteControllableTarget.h:
* Source/JavaScriptCore/inspector/remote/cocoa/RemoteInspectorCocoa.mm:
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorSocket.cpp:
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/WebCore/inspector/InspectorBackendClient.h:
(WebCore::InspectorBackendClient::allowRemoteInspectionToPageDirectly const): Deleted.
* Source/WebCore/inspector/InspectorFrontendClientLocal.h:
* Source/WebCore/page/Page.cpp:
(WebCore::m_mediaSessionManagerFactory):
* Source/WebKitLegacy/WebCoreSupport/LegacyWebPageDebuggable.cpp: Added.
(LegacyWebPageDebuggable::LegacyWebPageDebuggable):
(LegacyWebPageDebuggable::create):
(LegacyWebPageDebuggable::name const):
(LegacyWebPageDebuggable::url const):
(LegacyWebPageDebuggable::hasLocalDebugger const):
(LegacyWebPageDebuggable::connect):
(LegacyWebPageDebuggable::disconnect):
(LegacyWebPageDebuggable::dispatchMessageFromRemote):
(LegacyWebPageDebuggable::setIndicating):
(LegacyWebPageDebuggable::setNameOverride):
(LegacyWebPageDebuggable::detachFromPage):
* Source/WebKitLegacy/WebCoreSupport/LegacyWebPageDebuggable.h: Added.
* Source/WebKitLegacy/WebCoreSupport/LegacyWebPageInspectorController.cpp: Added.
(PageTarget::identifier):
(PageTarget::PageTarget):
(FrameTarget::identifier):
(FrameTarget::FrameTarget):
(LegacyWebPageInspectorController::create):
(LegacyWebPageInspectorController::LegacyWebPageInspectorController):
(LegacyWebPageInspectorController::frameCreated):
(LegacyWebPageInspectorController::frameDestroying):
(LegacyWebPageInspectorController::pageDestroying):
(LegacyWebPageInspectorController::addTarget):
(LegacyWebPageInspectorController::removeTarget):
(LegacyWebPageInspectorController::connectFrontend):
(LegacyWebPageInspectorController::disconnectFrontend):
(LegacyWebPageInspectorController::hasLocalFrontend const):
(LegacyWebPageInspectorController::disconnectAllFrontends):
(LegacyWebPageInspectorController::dispatchMessageFromFrontend):
(LegacyWebPageInspectorController::sendMessageToInspectorFrontend):
* Source/WebKitLegacy/WebCoreSupport/LegacyWebPageInspectorController.h: Added.
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/cf/WebCoreSupport/WebInspectorClientCF.cpp:
(WebInspectorClient::sendMessageToFrontend):
* Source/WebKitLegacy/ios/WebCoreSupport/WebInspectorClientIOS.mm:
(WebInspectorFrontendClient::WebInspectorFrontendClient):
* Source/WebKitLegacy/mac/WebCoreSupport/WebInspectorClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebInspectorClient.mm:
(WebInspectorClient::openLocalFrontend):
(WebInspectorFrontendClient::WebInspectorFrontendClient):
(WebInspectorFrontendClient::sendMessageToBackend):
(-[WebInspectorWindowController destroyInspectorView]):
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(+[WebFrame _createFrameWithPage:frameName:frameView:ownerElement:]):
(+[WebFrame _createMainFrameWithPage:frameName:frameView:]):
(-[WebFrame _clearCoreFrame]):
* Source/WebKitLegacy/mac/WebView/WebFrameInternal.h:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _commonInitializationWithFrameName:groupName:]):
(-[WebView initSimpleHTMLDocumentWithStyle:frame:preferences:groupName:]):
(-[WebView _close]):
(-[WebView allowsRemoteInspection]):
(-[WebView setAllowsRemoteInspection:]):
(-[WebView inspectorController]):
* Source/WebKitLegacy/mac/WebView/WebViewData.h:
* Source/WebKitLegacy/mac/WebView/WebViewInternal.h:

Canonical link: https://commits.webkit.org/304467@main
